tests/backup/backup_stork_test.go

```diff
@@ -0,0 +1,569 @@
+			restoreNames = append(restoreNames, restoreName)
+		})
+		Step("Deleting new admin namespace", func() {
+			log.Info("Deleting namespace - %v", newAdminNamespace)
+			err := DeleteAppNamespace(newAdminNamespace)
+			log.FailOnError(err, "Unable to delete admin namespace")
+			log.InfoD("Namespace - %v - deleted successfully", newAdminNamespace)
+		})
+		Step("Restoring backup of multiple namespaces after admin namespace removal", func() {
+			log.InfoD("Restoring backup of multiple namespaces - Admin namespace will be recreated")
+			selectedBkpNamespaceMapping = make(map[string]string)
+			multipleRestoreMapping = make(map[string]string)
+			restoreName = fmt.Sprintf("%s-%v", restoreNamePrefix, time.Now().Unix())
+			for _, namespace := range bkpNamespaces {
+				selectedBkpNamespaceMapping[namespace] = namespace
+			}
+			log.InfoD("Selected application namespaces to restore: [%v]", bkpNamespaces)
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = CreateRestoreWithCRValidation(restoreName, backupName, selectedBkpNamespaceMapping, SourceClusterName, orgID, ctx, make(map[string]string))
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creating restore [%s]", restoreName))
+
+			// Restore to custom namespace
+			for _, namespace := range bkpNamespaces {
+				restoredNameSpace := fmt.Sprintf("%s-%v", backupName, time.Now().Unix())
+				multipleRestoreMapping[namespace] = restoredNameSpace
+			}
+			log.Infof("Custom restore map %v", multipleRestoreMapping)
+			customRestoreName := fmt.Sprintf("%s-%v", "multiple-application", time.Now().Unix())
+			err = CreateRestoreWithCRValidation(customRestoreName, backupName, multipleRestoreMapping, SourceClusterName, orgID, ctx, make(map[string]string))
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creating restore [%s]", restoreName))
+		})
+		Step("Deleting new admin namespace again", func() {
+			log.Info("Deleting namespace - %v", newAdminNamespace)
+			err := DeleteAppNamespace(newAdminNamespace)
+			log.FailOnError(err, "Unable to delete admin namespace")
+			log.InfoD("Namespace - %v - deleted successfully", newAdminNamespace)
+		})
+		Step("Taking backup of multiple namespaces after admin namespace removal", func() { // This step should fail
+			backupName = fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+			restoreName = fmt.Sprintf("%s-%v", RestoreNamePrefix, time.Now().Unix())
+			log.InfoD(fmt.Sprintf("Taking backup of multiple namespaces [%v]", bkpNamespaces))
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = CreateBackupWithCRValidation(backupName, SourceClusterName, bkpLocationName, backupLocationUID, bkpNamespaces, labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+			dash.VerifyFatal(strings.Contains(err.Error(), "CR"), true, fmt.Sprintf("Backup creation failed due to non existing namespace [%s]. Error : %s", newAdminNamespace, err.Error()))
```

Comment:
Can we add a more elaborate error string instead of just `CR` ?

Reply:
Done